### PR TITLE
Provisioning Documentation Updates

### PR DESCRIPTION
**Pull request by michaelgugino**
Updating provisioning documents.

Also moved some steps from build_ami.yml to a new
play for better reusability.

### CODE REVIEW

File: playbooks/aws/README.md [hunk 1]
```diff
@@ -41,82 +53,38 @@ The newly added playbooks are the following:
...
-# This will ensure these users are created
-#openshift_aws_users:
-#- key_name: myuser_key
-#  username: myuser
-#  pub_key: |
-#         ssh-rsa AAAA
+# Minimum mandatory provisioning variables.  See provisioning_vars.yml.example.
+# for more information.
+openshift_deployment_type: # 'origin' or 'openshift-enterprise'
+openshift_release: # example: v3.6
```
Comment (kwoodson) [inline playbooks/aws/README.md]: probably should put in 3.7

File: playbooks/aws/README.md [hunk 2]
```diff
@@ -41,82 +53,38 @@ The newly added playbooks are the following:
...
+openshift_pkg_version: # example: -3.6.0
```
Comment (kwoodson) [inline playbooks/aws/README.md]: same as above, 3.7.0

File: playbooks/aws/README.md [hunk 3]
```diff
@@ -195,13 +143,13 @@ Once this playbook completes, the cluster masters should be installed and config
 
 #### Step 5
 
-Now that we have a cluster deployed it will be more interesting to create some node types.  This can be done easily with the following playbook:
+Now that we the cluster masters deployed, we need to deploy our infrastructure and compute nodes:
```
Comment (kwoodson) [inline playbooks/aws/README.md]: Fix sentence.
```
we the
```

File: playbooks/aws/provisioning-inventory.example.ini [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+# openshift_deployment_type is required for installation
+openshift_deployment_type=origin
+
+openshift_master_bootstrap_enabled=True
+
+openshift_hosted_router_wait=False
+openshift_hosted_registry_wait=False
+
+# Repository for installation
+openshift_additional_repos=[{'name': 'openshift-repo', 'id': 'openshift-repo',  'baseurl': 'https://mirror.openshift.com/enterprise/enterprise-3.6/latest/x86_64/os/', 'enabled': 'yes', 'gpgcheck': 0, 'sslverify': 'no', 'sslclientcert': '/var/lib/yum/client-cert.pem', 'sslclientkey': '/var/lib/yum/client-key.pem', 'gpgkey': 'https://mirror.ops.rhcloud.com/libra/keys/RPM-GPG-KEY-redhat-release https://mirror.ops.rhcloud.com/libra/keys/RPM-GPG-KEY-redhat-beta https://mirror.ops.rhcloud.com/libra/keys/RPM-GPG-KEY-redhat-openshifthosted'}]
```
Comment (kwoodson) [inline playbooks/aws/provisioning-inventory.example.ini]: If we want to keep this as an example I'm fine with this.  The QE team expected this to work out-of-the-box.  This is not the case as the AMI will require the ssl client cert information. 

This should also point to 3.7

File: playbooks/aws/provisioning_vars.yml.example [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+# Common/Cluster Variables #
+# ------------------------ #
+# Variables in this section affect all areas of the cluster
+
+# Deployment type must be specified.
+openshift_deployment_type: # 'origin' or 'openshift-enterprise'
+
+# openshift_release must be specified.  Use whatever version of openshift
+# that is supported by openshift-ansible that you wish.
+openshift_release: # v3.6
```
Comment (kwoodson) [inline playbooks/aws/provisioning_vars.yml.example]: v3.7

File: playbooks/aws/provisioning_vars.yml.example [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+# This will be dependent on the version provided by the yum repository
+openshift_pkg_version: # -3.6.0
```
Comment (kwoodson) [inline playbooks/aws/provisioning_vars.yml.example]: -3.7.0

File: playbooks/aws/provisioning-inventory.example.ini (hunk 4 above)
Comment (michaelgugino) [reply playbooks/aws/provisioning-inventory.example.ini]: I'm not sure what we should do for this.  If we're using origin by default, we should probably just remove this.

**Pull request merged**